domain/application/state/application.go

```diff
@@ -3014,6 +3016,282 @@ WHERE application_uuid = $applicationID.uuid;
+	if err != nil {
+		return internalerrors.Errorf("preparing select container type id query: %w", err)
+	}
+
+	insertConstraintsQuery := `INSERT INTO "constraint"(*) VALUES ($setConstraint.*)`
+	insertConstraintsStmt, err := st.Prepare(insertConstraintsQuery, setConstraint{})
+	if err != nil {
+		return internalerrors.Errorf("preparing insert constraints query: %w", err)
+	}
+
+	insertConstraintTagsQuery := `INSERT INTO constraint_tag(*) VALUES ($setConstraintTag.*)`
+	insertConstraintTagsStmt, err := st.Prepare(insertConstraintTagsQuery, setConstraintTag{})
+	if err != nil {
+		return internalerrors.Errorf("preparing insert constraint tags query: %w", err)
+	}
+
+	insertConstraintSpacesQuery := `INSERT INTO constraint_space(*) VALUES ($setConstraintSpace.*)`
+	insertConstraintSpacesStmt, err := st.Prepare(insertConstraintSpacesQuery, setConstraintSpace{})
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	insertConstraintZonesQuery := `INSERT INTO constraint_zone(*) VALUES ($setConstraintZone.*)`
+	insertConstraintZonesStmt, err := st.Prepare(insertConstraintZonesQuery, setConstraintZone{})
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	insertAppConstraintsQuery := `INSERT INTO application_constraint(*) VALUES ($setApplicationConstraint.*)`
+	insertAppConstraintsStmt, err := st.Prepare(insertAppConstraintsQuery, setApplicationConstraint{})
+	if err != nil {
+		return internalerrors.Errorf("preparing insert application constraints query: %w", err)
+	}
+
+	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		var containerTypeID containerTypeID
+		if cons.Container != nil {
+			err := tx.Query(ctx, selectContainerTypeIDStmt, containerTypeVal{Value: string(*cons.Container)}).Get(&containerTypeID)
+			if errors.Is(err, sql.ErrNoRows) {
+				return applicationerrors.InvalidApplicationConstraints
+			}
+			if err != nil {
+				return internalerrors.Capture(err)
+			}
+		}
+
+		constraints := encodeConstraints(constraintUUID.String(), cons, containerTypeID.ID)
+
+		if err := tx.Query(ctx, insertConstraintsStmt, constraints).Run(); err != nil {
+			return internalerrors.Capture(err)
+		}
+
+		if cons.Tags != nil {
+			for _, tag := range *cons.Tags {
+				constraintTag := setConstraintTag{ConstraintUUID: constraintUUID.String(), Tag: tag}
+				if err := tx.Query(ctx, insertConstraintTagsStmt, constraintTag).Run(); err != nil {
+					return internalerrors.Capture(err)
+				}
+			}
+		}
+
+		if cons.Spaces != nil {
+			for _, space := range *cons.Spaces {
+				constraintSpace := setConstraintSpace{ConstraintUUID: constraintUUID.String(), Space: space}
+				err := tx.Query(ctx, insertConstraintSpacesStmt, constraintSpace).Run()
+				// Space is not found.
+				if internaldatabase.IsErrConstraintForeignKey(err) {
+					return applicationerrors.InvalidApplicationConstraints
+				}
+				if err != nil {
+					return internalerrors.Capture(err)
```

Comment:
Just use `errors` not `internalerrors`

Reply:
As a matter of fact we had imported twice our internal/errors package (once named the other wasn't), so I removed the double import and renamed all of them.